### PR TITLE
MEMNoGui: --install-mods supports file with .mfl extension that contains list of files to install

### PR DESCRIPTION
**Pull request by Mgamerz**
I'm trying to build some more automated installation features for MassEffectModderNoGui. I have compiled and tested using a '.mfl' file with --install-mods that you pass as --input parameter. This can be used to avoid having to stage files in a specific directory (something I should have probably done for in ALOTInstaller).

### CODE REVIEW

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 1]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
-    auto files = QDir(inputDir, "*.mem",
-                      QDir::SortFlag::Name | QDir::SortFlag::IgnoreCase,
-                      QDir::Files | QDir::NoDotAndDotDot | QDir::NoSymLinks).entryInfoList();
     QStringList modFiles;
-    foreach (QFileInfo file, files)
-    {
-        modFiles.push_back(file.absoluteFilePath());
-    }
 
+    if (QDir(inputDir).exists()){
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: formatting, wrong placement of '{'

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 2]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
+        auto files = QDir(inputDir, "*.mem",
+                          QDir::SortFlag::Name | QDir::SortFlag::IgnoreCase,
+                          QDir::Files | QDir::NoDotAndDotDot | QDir::NoSymLinks).entryInfoList();
+        foreach (QFileInfo file, files)
+        {
+            modFiles.push_back(file.absoluteFilePath());
+        }
+    } else if (QFile(inputDir).exists()){
+        QFileInfo fi(inputDir);
+        QString ext = fi.completeSuffix();  // ext = "tar.gz")
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: is that comment related?

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 3]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
+        if (ext == "mfl") {
+            // MEM File List
+            QFile inputFile(inputDir);
+            if (inputFile.open(QIODevice::ReadOnly))
+            {
+                QTextStream in(&inputFile);
+                while (!in.atEnd())
+                {
+                   QString line = in.readLine();
+                   if (QFile(line).exists()){
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: formatting, wrong placement of '{'

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 4]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
 
+    if (QDir(inputDir).exists()){
+        auto files = QDir(inputDir, "*.mem",
+                          QDir::SortFlag::Name | QDir::SortFlag::IgnoreCase,
+                          QDir::Files | QDir::NoDotAndDotDot | QDir::NoSymLinks).entryInfoList();
+        foreach (QFileInfo file, files)
+        {
+            modFiles.push_back(file.absoluteFilePath());
+        }
+    } else if (QFile(inputDir).exists()){
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: formatting, wrong placement of '{'

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 5]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
+            {
+                QTextStream in(&inputFile);
+                while (!in.atEnd())
+                {
+                   QString line = in.readLine();
+                   if (QFile(line).exists()){
+                       modFiles.push_back(line);
+                   }
+                }
+                inputFile.close();
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: there is no need call close(), it will close while destruction of QFile from stack

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 6]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
+                          QDir::SortFlag::Name | QDir::SortFlag::IgnoreCase,
+                          QDir::Files | QDir::NoDotAndDotDot | QDir::NoSymLinks).entryInfoList();
+        foreach (QFileInfo file, files)
+        {
+            modFiles.push_back(file.absoluteFilePath());
+        }
+    } else if (QFile(inputDir).exists()){
+        QFileInfo fi(inputDir);
+        QString ext = fi.completeSuffix();  // ext = "tar.gz")
+        if (ext == "mfl") {
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: formatting, wrong placement of '{'

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp (hunk 2 above)
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp:526]: I suggest add conversion to lower case here, like fi.completeSuffix().toLower()

File: MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp [hunk 7]
```diff
@@ -511,15 +511,36 @@ bool CmdLineTools::InstallMods(MeType gameId, QString &inputDir,
...
+            // MEM File List
+            QFile inputFile(inputDir);
+            if (inputFile.open(QIODevice::ReadOnly))
+            {
+                QTextStream in(&inputFile);
+                while (!in.atEnd())
+                {
+                   QString line = in.readLine();
```
Comment (aquadran) [inline MassEffectModder/MassEffectModder/CmdLine/CmdLineTools.cpp]: please add change path delimiter to unified '/', like this: line.replace(QChar('\\'), QChar('/'));

**Review by aquadran — CHANGES_REQUESTED** [review]

**Comment by Mgamerz** [comment]
Added another commit resolving the above requests.
